src/App.js

```diff
@@ -1,5 +1,109 @@
+import { Console } from '@woowacourse/mission-utils';
+import {
+  CONSOLE_MESSAGE,
+  DEFAULT_SEPARATOR,
+  SPECIAL_CHARACTERS,
+} from './constant.js';
+import {
+  isNumber,
+  isPositiveNumber,
+  errorString,
+  escapeRegex,
+} from './util.js';
+
 class App {
-  async run() {}
+  async run() {
+    const input = await Console.readLineAsync(CONSOLE_MESSAGE.INPUT);
+
+    const [customSep, extractedStr] = this.extractCustomSeparator(input);
+
+    const splittedString = this.splitWithSeparator(
+      extractedStr,
+      DEFAULT_SEPARATOR,
+      customSep,
+    );
+
+    const sum = this.sumAllString(splittedString);
+
+    Console.print(`결과 : ${sum}`);
+  }
+
+  extractCustomSeparator(str) {
+    const [first, rest] = str.split('\\n');
+
+    if (rest && !first.startsWith('//')) {
+      throw Error(errorString(CONSOLE_MESSAGE.SEPARATOR_FORM_ERROR));
+    }
+
+    if (first.startsWith('//')) {
+      const customSep = first.slice(2);
+
+      this.validateSeparator(customSep);
+
+      return [customSep, rest];
+    }
+
+    return [null, first];
+  }
+
+  validateSeparator(customSep) {
+    if (DEFAULT_SEPARATOR.includes(customSep)) {
+      throw Error(errorString(CONSOLE_MESSAGE.SEPARATOR_DUPLICATE_ERROR));
+    }
+
+    if (customSep.length !== 1) {
+      throw Error(errorString(CONSOLE_MESSAGE.SEPARATOR_LENGTH_ERROR));
+    }
+
+    if (customSep === ' ') {
+      throw Error(errorString(CONSOLE_MESSAGE.SEPARATOR_EMPTY_ERROR));
+    }
+  }
+
+  splitWithSeparator(str, separator, customSep) {
+    if (!str || str.trim() === '') {
+      return ['0'];
+    }
+
+    const allSeparator = separator;
```

Comment:
```suggestion
    const allSeparator = [...separator];
```
요렇게 스프레드 연산자로 배열을 복사하는 것이 더 좋아보입니다! 둘다 같은 배열을 '참조'하게 되기 때문에 가변성에 의해 위험해질 것 같아요!

Reply:
네네 이 부분도 제가 놓친 부분이더라구요. 감사합니당